phoenicis-repository/src/main/java/org/phoenicis/repository/repositoryTypes/MergeableRepository.java

```diff
@@ -107,12 +101,46 @@ protected ApplicationDTO mergeApplications(ApplicationDTO leftApplication, Appli
         final List<ResourceDTO> resources = mergeListOfDtos(leftApplication.getResources(),
                 rightApplication.getResources(), ResourceDTO::getName, ResourceDTO.nameComparator());
 
-        final Set<URI> mergeMiniaturesSet = new HashSet<URI>(leftApplication.getMiniatures());
-        mergeMiniaturesSet.addAll(rightApplication.getMiniatures());
+        final List<URI> mergeMiniatures = mergeMiniatures(leftApplication.getMiniatures(),
+                rightApplication.getMiniatures());
 
         return new ApplicationDTO.Builder().withName(leftApplication.getName()).withResources(resources)
                 .withScripts(scripts).withDescription(leftApplication.getDescription())
-                .withIcon(leftApplication.getIcon()).withMiniatures(new ArrayList<>(mergeMiniaturesSet)).build();
+                .withIcon(leftApplication.getIcon()).withMiniatures(mergeMiniatures).build();
+    }
+
+    /**
+     * Takes two lists of {@link URI}s leading to miniature images and merges them into a single list.
+     * During the merging all duplicates are removed, while the miniatures inside <code>leftMiniatures</code> have a higher priority
+     *
+     * @param leftMiniatures  The first list of miniature {@link URI}s
+     * @param rightMiniatures The first list of miniature {@link URI}s
+     * @return A list containing the merged miniature {@link URI}s from both <code>leftMiniatures</code> and <code>rightMiniatures</code>
+     */
+    protected List<URI> mergeMiniatures(List<URI> leftMiniatures, List<URI> rightMiniatures) {
+        HashMap<String, URI> mergedMiniatures = new HashMap<>();
+
+        /*
+         * Concatenate the both lists with the left list in front of the right one
+         */
+        List<URI> miniatures = new ArrayList<>(leftMiniatures);
+        miniatures.addAll(rightMiniatures);
+
+        /*
+         * Remove duplicates
+         */
+        for (URI miniatureUri : miniatures) {
+            try (InputStream inputStream = miniatureUri.toURL().openStream()) {
+                String checksum = DigestUtils.md5Hex(inputStream);
```

Comment:
These `URI`s should be local files. I'm only converting them to `URL` objects, because it's quite easy to create an `InputStream` from an `URL`. 
When we later add an `OnlineRepository`, that uses REST for example, I would also suggest to download the full repository, save it locally and update it at startup, like we do for our `GitRepository`.
Because we've local files behind the `URI` objects this should be quite fast, right?